torchx/mix.exs

```diff
@@ -43,4 +45,119 @@ defmodule Torchx.MixProject do
       source_url: @source_url
     ]
   end
+
+  defp compilers do
+    if System.get_env("LIBTORCH_DIR"), do: [:elixir_make], else: [:torchx, :elixir_make]
+  end
+
+  defp aliases do
+    [
+      "compile.torchx": &compile/1
+    ]
+  end
+
+  defp compile(args) do
+    cache_dir = Path.join(__DIR__, "cache")
+    # TODO: This directory name must also include the libtorch version.
+    # Otherwise if someone depends on torchx as a git dependency, they
+    # can upgrade the version and we will think it is cached but it is actually old.
+    libtorch_dir = Path.join(cache_dir, "libtorch_#{@default_libtorch_version}")
+
+    if "--force" in args do
+      File.rm_rf!(cache_dir)
+    end
+
+    if File.dir?(libtorch_dir) do
+      {:ok, []}
+    else
+      install_libtorch(cache_dir, libtorch_dir)
+    end
+  end
+
+  defp install_libtorch(cache_dir, libtorch_dir) do
+    File.mkdir_p!(cache_dir)
+
+    libtorch_zip =
+      System.get_env(
+        "LIBTORCH_ZIP",
+        Path.join(cache_dir, "libtorch_#{@default_libtorch_version}.zip")
+      )
+
+    unless File.exists?(libtorch_zip) do
+      # Download libtorch
+
+      # Sanity check for future changes or updates on the default libtorch version
+      if @default_libtorch_version != "1_9_1_cpu",
+        do: raise("ensure the download URLs match the default libtorch version")
+
+      url =
+        case :os.type() do
+          {:unix, :linux} ->
+            "https://download.pytorch.org/libtorch/cpu/libtorch-cxx11-abi-shared-with-deps-1.9.1%2Bcpu.zip"
+
+          {:unix, :darwin} ->
+            # MacOS
+            "https://download.pytorch.org/libtorch/cpu/libtorch-macos-1.9.1.zip"
+
+          os ->
+            raise "OS #{os} is not supported"
+        end
+
+      download!(url, libtorch_zip)
+    end
+
+    # Unpack libtorch to libtorch_dir
+
+    {:ok, _} =
+      libtorch_zip |> String.to_charlist() |> :zip.unzip(cwd: String.to_charlist(cache_dir))
+
+    unpack_dir = Path.join(cache_dir, "libtorch")
+
+    unless unpack_dir == libtorch_dir do
```

Comment:
This can never be true, can it?

Reply:
This made more sense when we always ran torchx.compile. Basically, we would always download and unpack to cache/libtorch, and this File.rename would then unpack to the specified directory if the env var wasn't a valid directory originally.

I'll remove this (and the rename) because now we either download and unpack specifically to cache/libtorch, or we just skip the whole download/unpacking thing